### ISSUE TITLE
Parsing schemaorg recipes

### ISSUE THREAD

**Issue by mainrs** [timothee-haudebourg/json-ld#34]
Hello!

I have trouble figuring out how to parse recipe data encoded as JSON-LD that follows the https://schema.org specification.

For example, I am trying to parse this document: https://developers.google.com/search/docs/advanced/structured-data/recipe#guided-example

But with the following code, it fails due to `thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Loc { value: Error { code: LoadingRemoteContextFailed, source: None }, source: None, metadata: () }', packages/jsonld/src/main.rs:10:81`.

```rust
use iref::IriBuf;
use json_ld::{Reference, NoLoader, Document, context, Object};
use serde_json::Value;

pub async fn parse_recipes(jsonld: impl AsRef<str>) -> Result<(), ()> {
    let jsonld = jsonld.as_ref();
    let doc: Value = serde_json::from_str(jsonld).unwrap();

    let mut loader = json_ld::reqwest::Loader::<Value>::new(|jsonld| serde_json::from_str(jsonld));
    let expanded_doc = doc.expand::<context::Json<Value>, _>(&mut loader).await.unwrap(); // <--- line 10

    let name_property = Reference::Id(IriBuf::new("http://xmlns.com/foaf/0.1/name").unwrap());

    for object in expanded_doc {
        if let Object::Node(node) = object.as_ref() {
            println!("node: {}", node.id().unwrap());
            for name in node.get(&name_property) {
                println!("name: {}", name.as_str().unwrap());
            }
        }
    }

    Ok(())
}

#[tokio::main]
async fn main() {
    let jsonld = std::fs::read_to_string("r1.json").unwrap();
    parse_recipes(&jsonld).await.unwrap();
}
```

Any help is appreciated!

**Comment by timothee-haudebourg** [comment]
The `reqwest` loader is a very basic HTTP client that will execute a `GET` query with the header `Accept: application/ld+json, application/json` to download a JSON-LD document.

Using the following commend, I checked the answer of `schema.org` on querying the JSON-LD context:
```
wget -d --header="Accept: application/ld+json, application/json" https://schema.org/
```
The response is not a JSON document, but an HTML document, that's why it fails. The response also contains a `Link` header the provides the URL to the actual JSON-LD context. A complete implementation of https://w3c.github.io/json-ld-api/#loaddocumentcallback should use this link to retreive the context, but as I said, the `reqwest` loader is for now a very basic loader implementation.

You can either write your own more robust loader, or wait for me to improve the `reqwest` loader (See #35).

**Comment by timothee-haudebourg** [comment]
Another solution would be to ship the `schema.org` JSON-LD context with your application and make your own loader that loads the shipped context when the URL is `https://schema.org`, instead of downloading it. In general this is preferable since it is more secure and avoid downloading a huge file (especially for `schema.org`) every time.

**Comment by mainrs** [comment]
Thanks for the help!

**Comment by timothee-haudebourg** [comment]
FYI this has been implemented in version 0.9.1.
